commands/build.go

```diff
@@ -321,12 +322,18 @@ func runBuild(ctx context.Context, dockerCli command.Cli, options buildOptions)
 	if err != nil {
 		return err
 	}
+
 	_, err = b.LoadNodes(ctx)
 	if err != nil {
 		return err
 	}
 	driverType := b.Driver
 
+	dockerUsingContainerdSnapshotter := false
+	if driverType == "docker" || driverType == "docker-container" {
```

Comment:
I don't think there is anything special for these drivers. For "docker" containerd support can be detected via feature flag like it is for other things not supported by graphdrivers.

For other drivers, I guess they are only affected if `--load` is set. If there is no `--load` then there is no reason to check a specific docker instance at build time, as it would be completely unused. If this feature has problems with other drivers as well on `--load` then I think we need to capture the image ID from moby side when we are doing `docker load`. So we load image into Moby, and it reports to us what is the ID of the image that was just imported (as it would be different in different Moby versions). It's quite weird though that the value of `--iidfile` depends on in `--load` is set or not.